### PR TITLE
[Fleet] add fleet-server-remote service account

### LINKED ISSUE #85747: [Fleet] Create new service account `elastic/fleet-server-remote` for remote agent connections
### Description

**Background**
As part of https://github.com/elastic/kibana/issues/104986 we will be adding the ability for Elastic Agents to send agent monitoring data to a remote Elasticsearch cluster. In order to achieve this, we will need a service token for the remote cluster which fleet server will use. 

This service account will need less permissions than the `elastic/fleet-server` as integration data will not be sent to remote clusters yet.

**Service Account Details**

Name: `elastic/fleet-server-remote`

Permissions:
-  for indices `logs-*`, `metrics-*` privileges "write", "create_index", "auto_configure"

### PR DESCRIPTION
**Pull request by juliaElastic**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? yes
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)? yes
- If submitting code, have you built your formula locally prior to submission with `gradle check`? yes
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)? yes
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

Resolves https://github.com/elastic/elasticsearch/issues/85747

The `elastic/fleet-server-remote` token would be used by Fleet Server to create API keys for that cluster to send those down to each Elastic Agent. Is the `manage_own_api_key` privilege sufficient for that? `elastic/fleet-server` has the same.

Other than that adding `logs-*` and `metrics-*` `"write", "create_index", "auto_configure"` privileges to send monitoring data.

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java [hunk 1]
```diff
@@ -150,13 +150,34 @@ final class ElasticServiceAccounts {
             null
         )
     );
+    private static final ServiceAccount FLEET_REMOTE_ACCOUNT = new ElasticServiceAccount(
+        "fleet-server-remote",
+        new RoleDescriptor(
+            NAMESPACE + "/fleet-server-remote",
+            new String[] { "manage_own_api_key" },
```
Comment (nchaulet) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java:157]: Do we need to add the `monitor` privileges? I think we currently add this to the api key we generate for agent

Comment (juliaElastic) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java]: done

**Comment by juliaElastic** [comment]
@elasticsearchmachine merge upstream

**Review by nchaulet — APPROVED** [review]
LGTM 🚀

**Pull request merged**